### PR TITLE
Check commands in commandhandler for args

### LINKED ISSUE #3798: [FEATURE] Add args present argument to command filter
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

There is currently no filter defined by us which simply checks if a command message has args, e.g. `start payload`

### Describe the solution you'd like

Add one argument to the command filter called `has_args` which checks if something comes after the command in the message.

### Describe alternatives you've considered

One can do this filter by themselves, but providing one would be nice

### Additional context

We could also start discussing if we want to provide some way to check the content of args, like a pattern match.

### PR DESCRIPTION
**Pull request by thatguylah**
Hi, so sorry, i now realize https://github.com/python-telegram-bot/python-telegram-bot/pull/3812 is closed, but i have finished amending changes in this new PR here. So sorry for the delay 🙏 this should resolve the comments in the PR previously. 

closes #3798, supersedes #3812

### CODE REVIEW

**Comment by thatguylah** [comment]
Adding @Bibo-Joshi  for review pls 🙏

File: telegram/ext/_commandhandler.py [hunk 1]
```diff
@@ -100,16 +108,22 @@ async def callback(update: Update, context: CallbackContext)
...
-    __slots__ = ("commands", "filters")
+    __slots__ = ("commands", "filters", "has_args")
 
     def __init__(
         self,
         command: SCT[str],
         callback: HandlerCallback[Update, CCT, RT],
         filters: Optional[filters_module.BaseFilter] = None,
         block: DVType[bool] = DEFAULT_TRUE,
+        has_args: Union[bool, None, int] = None,
```
Comment (harshil21) [inline telegram/ext/_commandhandler.py:126]:
```suggestion
        has_args: Optional[bool, int] = None,
```

File: telegram/ext/_commandhandler.py [hunk 2]
```diff
@@ -126,6 +140,33 @@ def __init__(
             filters if filters is not None else filters_module.UpdateType.MESSAGES
         )
 
+        self.has_args: Union[bool, None, int] = has_args
```
Comment (harshil21) [inline telegram/ext/_commandhandler.py:143]:
```suggestion
        self.has_args: Optional[bool, int] = has_args
```

File: telegram/ext/_commandhandler.py [hunk 3]
```diff
@@ -126,6 +140,33 @@ def __init__(
...
+
+    def _check_correct_args(self, args: List[str]) -> Optional[bool]:
+        """Determines whether the args are correct for this handler. Implemented in check_update().
+        Args:
+            args (:obj:`list`): The args for the handler.
+        Returns:
+            :obj:`bool`: Whether the args are valid for this handler.
+        """
+
```
Comment (harshil21) [inline telegram/ext/_commandhandler.py:152]:
```suggestion
```
remove blank line after method docstring

**Review by harshil21 — CHANGES_REQUESTED** [review]
Thanks for the PR. I see that you've not fully resolved Hinrich's review comments, specifically the wording in the docstrings (https://github.com/python-telegram-bot/python-telegram-bot/pull/3812#discussion_r1276098190 & https://github.com/python-telegram-bot/python-telegram-bot/pull/3812#discussion_r1276102232). Other than that, logic looks good and simple!

File: tests/ext/test_commandhandler.py [hunk 4]
```diff
@@ -181,7 +181,11 @@ async def test_basic(self, app, command):
 
     @pytest.mark.parametrize(
         "cmd",
-        ["way_too_longcommand1234567yes_way_toooooooLong", "ïñválídletters", "invalid #&* chars"],
+        [
+            "way_too_longcommand1234567yes_way_toooooooLong",
+            "ïñválídletters",
+            "invalid #&* chars",
+        ],
```
Comment (Bibo-Joshi) [inline tests/ext/test_commandhandler.py:188]: Is this change related to the PR? If not, please revert

File: tests/ext/test_commandhandler.py [hunk 5]
```diff
@@ -259,3 +263,31 @@ async def test_context_multiple_regex(self, app, command):
...
+
+        assert is_match(handler_int_one, make_command_update("/test helloworld", bot=bot))
+        assert not is_match(handler_int_one, make_command_update("/test hello world", bot=bot))
+        assert not is_match(handler_int_one, make_command_update("/test", bot=bot))
+
+        assert is_match(handler_int_two, make_command_update("/test hello world", bot=bot))
+        assert not is_match(handler_int_two, make_command_update("/test helloworld", bot=bot))
+        assert not is_match(handler_int_two, make_command_update("/test", bot=bot))
+
+        # Test for ValueError when has_args is a negative integer
```
Comment (Bibo-Joshi) [inline tests/ext/test_commandhandler.py:289]: Please move this to a separate test function

File: telegram/ext/_commandhandler.py (hunk 1 above)
Comment (thatguylah) [reply telegram/ext/_commandhandler.py:126]: Hi, i tried implementing your suggestion, but in return i am failing precommit hook `mypy-ptb` 

image

Suggest to either keep `Union[bool,None,int]` or change to `Optional[Union[bool, int]]` what do you think?

File: telegram/ext/_commandhandler.py (hunk 3 above)
Comment (thatguylah) [reply telegram/ext/_commandhandler.py:152]: Done.

File: telegram/ext/_commandhandler.py [hunk 6]
```diff
@@ -126,6 +140,33 @@ def __init__(
...
+        has_args_none = self.has_args is None
+        has_args_true_args_present = self.has_args is True and args
+        has_args_false_args_absent = self.has_args is False and not args
+        has_args_num = isinstance(self.has_args, int) and len(args) == self.has_args
+        has_args_num_negative_int = isinstance(self.has_args, int) and self.has_args < 0
+
+        if has_args_num_negative_int:
```
[the remark this answers is not in the document]
Comment (thatguylah) [reply telegram/ext/_commandhandler.py:159]: Done.

File: telegram/ext/_commandhandler.py [hunk 7]
```diff
@@ -126,6 +140,33 @@ def __init__(
...
+            raise ValueError("CommandHandler argument has_args cannot be a negative integer")
+        if (
+            (has_args_none)
```
[the remark this answers is not in the document]
Comment (thatguylah) [reply telegram/ext/_commandhandler.py:162]: Done.

File: tests/ext/test_commandhandler.py (hunk 4 above)
Comment (thatguylah) [reply tests/ext/test_commandhandler.py:188]: Reverted

File: tests/ext/test_commandhandler.py (hunk 5 above)
Comment (thatguylah) [reply tests/ext/test_commandhandler.py:289]: Done.

File: telegram/ext/_commandhandler.py (hunk 1 above)
Comment (harshil21) [reply telegram/ext/_commandhandler.py]: oh huh, didn't know that. Ok you can keep as it is then

**Review by harshil21 — APPROVED** [review]

**Review by Bibo-Joshi — APPROVED** [review]
Thank you for the nice contribution :)
